### DISCUSSION TITLE
Is there a way to differentiate between different keys in the `useOverlay()` hook?

### DISCUSSION THREAD

**Discussion by MoistCode** [adobe/react-spectrum#1992] [Q&A] [answered]
I am using the different overlay hooks to create a popover dialog that happens to contain a date picker. The behavior I want is:

1. If the user clicks outside of the trigger element, apply the dates.
2. If the user presses the "Escape" key, close the popover and do nothing with the dates.

My question is, is there a way to determine if the user press "Escape" or not. The `onClose()` handler is called but since the event object isn't passed through, I'm unable to check this.

https://github.com/adobe/react-spectrum/blob/main/packages/@react-aria/overlays/src/useOverlay.ts#L111-L117

**Comment by snowystinger** [comment, marked as answer]
This seems reasonable, sorry for the delay. We'd probably only want to expose it to the aria-hooks to start. Would you mind opening an issue for it? Feel free to contribute it as well (faster way to see the feature you need)
